test/document.js

```diff
@@ -290,13 +290,13 @@ describe('serialize document', function() {
   it("doesn't serialize unsupported types", function() {
     assert.throws(() => {
       firestore.doc('collectionId/documentId').set({foo: undefined});
-    }, /Cannot use custom type "undefined" as a Firestore type./);
+    }, /Invalid use of type "undefined" as a Firestore argument./);
 
     assert.throws(() => {
       firestore
         .doc('collectionId/documentId')
         .set({foo: Firestore.FieldPath.documentId()});
-    }, /Cannot use "FieldPath" as a Firestore type./);
+    }, /Cannot use object of type "FieldPath" as a Firestore value./);
```

Comment:
FWIW, I'd kinda' expect to have  a test case here that hits the new "doesn't support serialization of JavaScript objects that were constructed using the 'new' operator."  (I know it's covered the query.where() test case elsewhere, but it seems much more likely users will use custom objects with set() and so that's the error message I'm most interested in seeing tested...

Reply:
Added this error case to this test as well. That should now give me 101% coverage, right? :)